### PR TITLE
Enable test reports for all tests and save reports

### PR DESCRIPTION
**Pull request by lwr20**
## Description
Causes test reports to be created from UT and FV test runs and puts them in `./report` for later analysis.

## Release Note
```release-note
None required
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -303,14 +304,14 @@ ut: vendor
...
-		$(CALICO_BUILD) sh -c 'WHAT=$(WHAT) SKIP=$(SKIP) ./run-uts'
+		$(CALICO_BUILD) sh -c 'WHAT=$(WHAT) SKIP=$(SKIP) GINKGO_ARGS="$(GINKGO_ARGS)" ./run-uts'
 
 .PHONY: fv
 ## Build and run the FV tests.
 GINKGO_FOCUS?=.*
 fv: tests/fv/fv.test image
 	@echo Running Go FVs.
-	cd tests/fv && ETCD_IMAGE=$(ETCD_IMAGE) HYPERKUBE_IMAGE=$(HYPERKUBE_IMAGE) CONTAINER_NAME=$(BUILD_IMAGE):latest-$(ARCH) PRIVATE_KEY=`pwd`/private.key ./fv.test -ginkgo.slowSpecThreshold 30 -ginkgo.focus $(GINKGO_FOCUS)
+	cd tests/fv && ETCD_IMAGE=$(ETCD_IMAGE) HYPERKUBE_IMAGE=$(HYPERKUBE_IMAGE) CONTAINER_NAME=$(BUILD_IMAGE):latest-$(ARCH) PRIVATE_KEY=`pwd`/private.key ./fv.test $(GINKGO_ARGS) -ginkgo.slowSpecThreshold 30 -ginkgo.focus $(GINKGO_FOCUS)
```
Comment (caseydavenport) [inline Makefile]: Might want to drop the `GINKGO_FOCUS` option if we're using GINKGO_ARGS instead.

File: report/README.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+// This directory is intended to hold the junit XML reports generated by fv and unit tests.
```
Comment (caseydavenport) [inline report/README.txt]: Do we really need to check this in?

I'd suggest we make the directory if it doesn't exist, and add it to `.gitignore`.

Comment (lwr20) [reply report/README.txt]: This is what we've done in other repos (node, felix, etc), but OK.  

Where would be the right place to make that directory?  Its referred to in the test code, so would I create it in there?   Or in the makefile?  Having the directory checked in at least means that even if people run the tests without using the makefile (i.e. directly running `ginkgo` or `fv.test` the directory will be there).

Comment (caseydavenport) [reply report/README.txt]: If we've done it elsewhere, then let's be consistent at least. SGTM

**Pull request merged**